$NetBSD: patch-aa,v 1.15 2010/12/13 16:03:21 tron Exp $

Fix build problem under Mac OS X where snprintf() is a macro.

--- src/serv.c.orig	2010-12-06 13:04:44.000000000 +0000
+++ src/serv.c	2010-12-13 15:32:42.000000000 +0000
@@ -438,7 +438,6 @@
 
 /* Creates html with the current session information.
  */
-#define tmp2 &http_buffer[strlen(http_buffer)], len-strlen(http_buffer)
 static char *
 peer_print_info (gnutls_session_t session, int *ret_length,
 		 const char *header)
@@ -512,11 +511,11 @@
 
   /* print session_id */
   gnutls_session_get_id (session, sesid, &sesid_size);
-  snprintf (tmp2, "\n<p>Session ID: <i>");
+  snprintf (&http_buffer[strlen(http_buffer)], len-strlen(http_buffer), "\n<p>Session ID: <i>");
   for (i = 0; i < sesid_size; i++)
-    snprintf (tmp2, "%.2X", sesid[i]);
-  snprintf (tmp2, "</i></p>\n");
-  snprintf (tmp2,
+    snprintf (&http_buffer[strlen(http_buffer)], len-strlen(http_buffer), "%.2X", sesid[i]);
+  snprintf (&http_buffer[strlen(http_buffer)], len-strlen(http_buffer), "</i></p>\n");
+  snprintf (&http_buffer[strlen(http_buffer)], len-strlen(http_buffer),
 	   "<h5>If your browser supports session resuming, then you should see the "
 	   "same session ID, when you press the <b>reload</b> button.</h5>\n");
 
@@ -530,7 +529,7 @@
 
     if (gnutls_server_name_get (session, dns, &dns_size, &type, 0) == 0)
       {
-	snprintf (tmp2, "\n<p>Server Name: %s</p>\n", dns);
+	snprintf (&http_buffer[strlen(http_buffer)], len-strlen(http_buffer), "\n<p>Server Name: %s</p>\n", dns);
       }
 
   }
@@ -541,7 +540,7 @@
 #ifdef ENABLE_SRP
   if (kx_alg == GNUTLS_KX_SRP)
     {
-      snprintf (tmp2, "<p>Connected as user '%s'.</p>\n",
+      snprintf (&http_buffer[strlen(http_buffer)], len-strlen(http_buffer), "<p>Connected as user '%s'.</p>\n",
 	       gnutls_srp_server_get_username (session));
     }
 #endif
@@ -549,7 +548,7 @@
 #ifdef ENABLE_PSK
   if (kx_alg == GNUTLS_KX_PSK)
     {
-      snprintf (tmp2, "<p>Connected as user '%s'.</p>\n",
+      snprintf (&http_buffer[strlen(http_buffer)], len-strlen(http_buffer), "<p>Connected as user '%s'.</p>\n",
 	       gnutls_psk_server_get_username (session));
     }
 #endif
@@ -557,7 +556,7 @@
 #ifdef ENABLE_ANON
   if (kx_alg == GNUTLS_KX_ANON_DH)
     {
-      snprintf (tmp2,
+      snprintf (&http_buffer[strlen(http_buffer)], len-strlen(http_buffer),
 	       "<p> Connect using anonymous DH (prime of %d bits)</p>\n",
 	       gnutls_dh_get_prime_bits (session));
     }
@@ -565,7 +564,7 @@
 
   if (kx_alg == GNUTLS_KX_DHE_RSA || kx_alg == GNUTLS_KX_DHE_DSS)
     {
-      snprintf (tmp2,
+      snprintf (&http_buffer[strlen(http_buffer)], len-strlen(http_buffer),
 	       "Ephemeral DH using prime of <b>%d</b> bits.<br>\n",
 	       gnutls_dh_get_prime_bits (session));
     }
@@ -576,7 +575,7 @@
   tmp = gnutls_protocol_get_name (gnutls_protocol_get_version (session));
   if (tmp == NULL)
     tmp = str_unknown;
-  snprintf (tmp2,
+  snprintf (&http_buffer[strlen(http_buffer)], len-strlen(http_buffer),
 	   "<TABLE border=1><TR><TD>Protocol version:</TD><TD>%s</TD></TR>\n",
 	   tmp);
 
@@ -587,44 +586,44 @@
 					  (session));
       if (tmp == NULL)
 	tmp = str_unknown;
-      snprintf (tmp2, "<TR><TD>Certificate Type:</TD><TD>%s</TD></TR>\n", tmp);
+      snprintf (&http_buffer[strlen(http_buffer)], len-strlen(http_buffer), "<TR><TD>Certificate Type:</TD><TD>%s</TD></TR>\n", tmp);
     }
 
   tmp = gnutls_kx_get_name (kx_alg);
   if (tmp == NULL)
     tmp = str_unknown;
-  snprintf (tmp2, "<TR><TD>Key Exchange:</TD><TD>%s</TD></TR>\n", tmp);
+  snprintf (&http_buffer[strlen(http_buffer)], len-strlen(http_buffer), "<TR><TD>Key Exchange:</TD><TD>%s</TD></TR>\n", tmp);
 
   tmp = gnutls_compression_get_name (gnutls_compression_get (session));
   if (tmp == NULL)
     tmp = str_unknown;
-  snprintf (tmp2, "<TR><TD>Compression</TD><TD>%s</TD></TR>\n", tmp);
+  snprintf (&http_buffer[strlen(http_buffer)], len-strlen(http_buffer), "<TR><TD>Compression</TD><TD>%s</TD></TR>\n", tmp);
 
   tmp = gnutls_cipher_get_name (gnutls_cipher_get (session));
   if (tmp == NULL)
     tmp = str_unknown;
-  snprintf (tmp2, "<TR><TD>Cipher</TD><TD>%s</TD></TR>\n", tmp);
+  snprintf (&http_buffer[strlen(http_buffer)], len-strlen(http_buffer), "<TR><TD>Cipher</TD><TD>%s</TD></TR>\n", tmp);
 
   tmp = gnutls_mac_get_name (gnutls_mac_get (session));
   if (tmp == NULL)
     tmp = str_unknown;
-  snprintf (tmp2, "<TR><TD>MAC</TD><TD>%s</TD></TR>\n", tmp);
+  snprintf (&http_buffer[strlen(http_buffer)], len-strlen(http_buffer), "<TR><TD>MAC</TD><TD>%s</TD></TR>\n", tmp);
 
   tmp = gnutls_cipher_suite_get_name (kx_alg,
 				      gnutls_cipher_get (session),
 				      gnutls_mac_get (session));
   if (tmp == NULL)
     tmp = str_unknown;
-  snprintf (tmp2, "<TR><TD>Ciphersuite</TD><TD>%s</TD></TR></p></TABLE>\n",
+  snprintf (&http_buffer[strlen(http_buffer)], len-strlen(http_buffer), "<TR><TD>Ciphersuite</TD><TD>%s</TD></TR></p></TABLE>\n",
 	   tmp);
 
   if (crtinfo)
     {
-      snprintf(tmp2, "<hr><PRE>%s\n</PRE>\n", crtinfo);
+      snprintf(&http_buffer[strlen(http_buffer)], len-strlen(http_buffer), "<hr><PRE>%s\n</PRE>\n", crtinfo);
       free (crtinfo);
     }
 
-  snprintf(tmp2, "<hr><P>Your HTTP header was:<PRE>%s</PRE></P>\n" HTTP_END, header);
+  snprintf(&http_buffer[strlen(http_buffer)], len-strlen(http_buffer), "<hr><P>Your HTTP header was:<PRE>%s</PRE></P>\n" HTTP_END, header);
 
   *ret_length = strlen (http_buffer);
 
